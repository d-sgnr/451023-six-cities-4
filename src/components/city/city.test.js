import React from "react";
import renderer from "react-test-renderer";
import City from "./city.jsx";

import {Provider} from "react-redux";
import configureStore from "redux-mock-store";

const mockStore = configureStore([]);

const city = {
  name: `Amsterdam`,
  coordinates: [52.373057, 4.892557],
};

it(`City should be rendered correctly when not active`, () => {
  const store = mockStore({
    city,
  });

  const tree = renderer.create(
      <Provider store={store}>
        <City
          city={city}
          isActive={false}
          onCityClick={() => {}}
        />
      </Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});

it(`City should be rendered correctly when active`, () => {
  const store = mockStore({
    city,
  });

  const tree = renderer.create(
      <Provider store={store}>
        <City
          city={city}
          isActive={true}
          onCityClick={() => {}}
        />
      </Provider>).toJSON();

  expect(tree).toMatchSnapshot();
});
