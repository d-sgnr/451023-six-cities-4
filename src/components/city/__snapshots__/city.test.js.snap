// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`City should be rendered correctly when active 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item tabs__item--active"
    href="#"
    onClick={[Function]}
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;

exports[`City should be rendered correctly when not active 1`] = `
<li
  className="locations__item"
>
  <a
    className="locations__item-link tabs__item "
    href="#"
    onClick={[Function]}
  >
    <span>
      Amsterdam
    </span>
  </a>
</li>
`;
